### PR TITLE
The merger does not conflict changes from other forks.

### PR DESCRIPTION
**Pull request by Moro4125**
I hope I understood correctly notes.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,4 @@ vendor
 .idea
 .project
 composer.phar
+composer.lock
```
Comment (glensc) [inline .gitignore]: don't make irrelevant changes to your PR

File: README.md [hunk 2]
```diff
@@ -98,12 +98,32 @@ RequireJS documentation.
...
+}
+```
+
 Creating a Component
 --------------------
 
 To set up a Component to be installed with Component Installer, have it
 `require` the package *robloach/component-installer* and set the `type` to
-*component*:
+*component*, but it does not necessarily:
```
Comment (glensc) [inline README.md]: this should be spelled as "but it is not necessary"

File: README.md [hunk 3]
```diff
@@ -237,6 +257,38 @@ define use of [html5shiv](https://github.com/aFarkas/html5shiv):
 }
 ```
 
+### Packages Without Component Support In *composer.json*
+
+Using [`extra`](https://getcomposer.org/doc/04-schema.md#extra)
+in *composer.json* allows use of Component Installer in packages that don't
+explicitly provide support for component, but does ship with their own *composer.json*.
```
Comment (glensc) [inline README.md]: should be spelled as "but do ship"

File: README.md [hunk 4]
```diff
@@ -237,6 +257,38 @@ define use of [html5shiv](https://github.com/aFarkas/html5shiv):
 }
 ```
 
+### Packages Without Component Support In *composer.json*
+
+Using [`extra`](https://getcomposer.org/doc/04-schema.md#extra)
+in *composer.json* allows use of Component Installer in packages that don't
+explicitly provide support for component, but does ship with their own *composer.json*. 
+Using `extra` with packages that ships with Component Installer, will override component's settings for that package.
```
Comment (glensc) [inline README.md]: same here: "packages that ship"

File: src/ComponentInstaller/Installer.php [hunk 5]
```diff
@@ -73,11 +72,23 @@ public function getComponentPath(PackageInterface $package)
         $name = $prettyName = $package->getPrettyName();
         if (strpos($prettyName, '/') !== false) {
             list($vendor, $name) = explode('/', $prettyName);
+			unset($vendor);
```
Comment (glensc) [inline src/ComponentInstaller/Installer.php]: you should keep existing indenting

File: src/ComponentInstaller/Installer.php [hunk 6]
```diff
@@ -107,15 +118,20 @@ public function getComponentDir()
...
-        return parent::removeCode($package);
+        parent::removeCode($package);
     }
 
     /**
      * Remove a Component's files from the Component directory.
+	 *
+	 * @param PackageInterface $package
+	 * @return bool
      */
```
Comment (glensc) [inline src/ComponentInstaller/Installer.php]: formatting break here too. stick to existing formatting

File: src/ComponentInstaller/Installer.php [hunk 7]
```diff
@@ -125,15 +141,19 @@ public function removeComponent(PackageInterface $package)
 
     /**
      * Before installing the Component, be sure its destination is clear first.
+	 *
+	 * @param PackageInterface $package
      */
```
Comment (glensc) [inline src/ComponentInstaller/Installer.php]: and here

File: src/ComponentInstaller/Installer.php [hunk 8]
```diff
@@ -157,6 +177,7 @@ public static function postAutoloadDump(Event $event)
 
         // Initialize and execute each process in sequence.
         foreach ($processes as $class) {
+			/** @var \ComponentInstaller\Process\Process $process */
```
Comment (glensc) [inline src/ComponentInstaller/Installer.php]: and here

File: src/ComponentInstaller/Process/BuildJsProcess.php [hunk 9]
```diff
@@ -34,6 +29,7 @@ public function process()
      *
      * @param array $packages
      *   An array of packages.
+	 * @return bool
```
Comment (glensc) [inline src/ComponentInstaller/Process/BuildJsProcess.php]: and here

File: src/ComponentInstaller/Resources/require.js [hunk 10]
```diff
@@ -1,5 +1,5 @@
 /** vim: et:ts=4:sw=4:sts=4
- * @license RequireJS 2.1.5 Copyright (c) 2010-2012, The Dojo Foundation All Rights Reserved.
+ * @license RequireJS 2.1.8 Copyright (c) 2010-2012, The Dojo Foundation All Rights Reserved.
```
Comment (glensc) [inline src/ComponentInstaller/Resources/require.js]: updating requirejs to new version, should be probably different PR

**Comment by glensc** [comment]
wrote some bunch of inline comments. those are not related to project, just general guidelines.

however actual project maintainer has his final words on this :)

hope this helps!

**Comment by RobLoach** [comment]
Seeing that the composer.lock needs to be in there likely means a loose version discrepancy. We can deal with that later. I've added #81...

In the mean time, all these changes look great! Well done, @glensc, @Moro4125 et al. This is a big push forwards.
